src/irc/core/modes.c

```diff
@@ -498,7 +500,82 @@ static void event_mode(IRC_SERVER_REC *server, const char *data,
 			parse_channel_modes(chanrec, nick, mode, TRUE);
 	}
 
-	g_free(params);
+	g_free(channel);
+	g_free(mode);
+}
+
+// Given <params> to a MODE command, parse them into the target of the modes (a
+// channel or nickname) and a string holding the modes and their parameters.
+//
+// ":" will be removed from any <trailing> parameter.
+//
+// Since the parsing combines several parameters into one string, the exact
+// position of <trailing> is not particularly important. This is not very RFC
+// correct, but it has historically worked.
+//
+// For example:
+//
+// "#channel +o test" gets parsed to: target = "#channel" and modes = "+o test"
+// "#channel :+o test" gets parsed to: target = "#channel" and modes = "+o test"
+// "#channel +o :test" gets parsed to: target = "#channel" and modes = "+o test"
+//
+// The caller must free the target and modes strings.
```

Comment:
A block level comment would make sense. The code style generally prefers `/* */` comments everywhere, even if it's single line